### PR TITLE
Resolve COR's issues, add authorization header using interceptor, remove unnecessary code

### PR DESCRIPTION
**Pull request by atifsaddique211f**

### CODE REVIEW

**Comment by travisspencer** [comment]
`packages.json` didn't get cleaned up

**Comment by travisspencer** [comment]
`favicon.ico` can be deleted IINM

File: server/server.js [hunk 1]
```diff
@@ -4,65 +4,69 @@ var url = require('url');
...
+    var responseHeaders = {
+        'Content-Type': 'application/json',
+    };
 
-  // Making the naive assumption that any OPTIONS request is the user-agent making a cross-site preflighted request.
-  if (request.method === 'OPTIONS') {
-    console.log("Accepting probable preflight request");
+    var origin = request.headers["origin"];
 
+    // Making the naive assumption that any OPTIONS request is the user-agent making a cross-site preflighted request.
```
Comment (travisspencer) [inline server/server.js]: this actually isn't naive or assumed. It's checked on line 21. If `origin` is present, this is a CORS OPTIONS request -- i.e., preflight. Comment can be removed.

Comment (atifsaddique211f) [reply server/server.js]: done

**Comment by travisspencer** [comment]
`.angular-cli.json` looks like it can be cleaned up a bit too. Sorry to nitpick so much, but this is open source, so it's important to put our best forward.

**Pull request merged**